fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/eventslifecycle/MixinInGameHud.java

```diff
@@ -0,0 +1,18 @@
+package net.fabricmc.fabric.mixin.eventslifecycle;
+
+import net.fabricmc.fabric.api.event.client.HudRenderCallback;
+import net.minecraft.client.gui.hud.InGameHud;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(InGameHud.class)
+public class MixinInGameHud {
+
+	@Inject(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/InGameHud;renderStatusEffectOverlay()V", shift = At.Shift.AFTER))
```

Comment:
the method might have conditional returns etc so this may be the most reliable point.

Reply:
It is just somewhere in front of these, which are at the end:
```java
GlStateManager.color4f(1.0F, 1.0F, 1.0F, 1.0F);
GlStateManager.disableLighting();
GlStateManager.enableAlphaTest();
```

This injection point is in front of the f3 hud though.

Reply:
The callback should be fired as close to the end of the method body as possible. "in front of the f3 hud" is a misleading statement, as, with the position of the current injection, huds drawn in the callback would render *behind* the f3 hud due to draw order.

Reply:
Shall it be moved before the three lines that I mentioned?

Reply:
I've moved this inject point, please tell me any opinions as I am not sure if this is the right choice.

Reply:
My main worry with the new injection point is it can easily silently move if mojang changes things

Reply:
Yes, one single `color4f` in the middle would alter the whole inject point

Reply:
One possible solution:
Inject to `GlStateManager.disableLighting();` instead and call `GlStateManager.color4f(1.0F, 1.0F, 1.0F, 1.0F)` again ourseleves, should not affect performance much because `color4f` will not apply if the values are same as the cached values.

Reply:
Or inject to `GlStateManager.disableLighting();` and use shift by opcodes.

Reply:
Moved the inject point again.